web_sdk/web_test_utils/lib/skia_client.dart

```diff
@@ -0,0 +1,399 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:convert';
+import 'dart:io';
+
+import 'package:crypto/crypto.dart';
+import 'package:path/path.dart' as path;
+import 'package:process/process.dart';
+
+import 'environment.dart';
+
+const String _kGoldctlKey = 'GOLDCTL';
+
+const String _skiaGoldHost = 'https://flutter-engine-gold.skia.org';
+const String _instance = 'flutter-engine';
+
+/// A client for uploading image tests and making baseline requests to the
+/// Flutter Gold Dashboard.
+class SkiaGoldClient {
+  /// Creates a [SkiaGoldClient] with the given [workDirectory].
+  ///
+  /// The [browserName] parameter is the name of the browser that generated the
+  /// screenshots.
+  SkiaGoldClient(this.workDirectory, { required this.browserName });
+
+  /// Whether the Skia Gold client is available and can be used in this
+  /// environment.
+  static bool get isAvailable => Platform.environment.containsKey(_kGoldctlKey);
+
+  /// The name of the browser running the tests.
+  final String browserName;
+
+  /// A controller for launching sub-processes.
+  final ProcessManager process = const LocalProcessManager();
+
+  /// A client for making Http requests to the Flutter Gold dashboard.
+  final HttpClient httpClient = HttpClient();
+
+  /// The local [Directory] for the current test context. In this directory, the
+  /// client will create image and JSON files for the `goldctl` tool to use.
+  final Directory workDirectory;
+
+  String get _tempPath => path.join(workDirectory.path, 'temp');
+  String get _keysPath => path.join(workDirectory.path, 'keys.json');
+  String get _failuresPath => path.join(workDirectory.path, 'failures.json');
+
+  /// Whether the `goldctl` tool has been initialized.
+  bool get isInitialized => _isInitialized;
+  bool _isInitialized = false;
```

Comment:
This could be easily confused with `authorized`, which is check via a method instead of a getter. Can `authorized` and `initialized` use the same pattern with some more docs? Either way is fine, but it might be easier to read if they were treated similarly.

Reply:
Sounds good. I'll change `clientIsAuthorized` to be similar to `isInitialized` and document both.